catalyst-gateway/bin/src/service/api/cardano/rbac/registrations_get/chain_info.rs

```diff
@@ -36,51 +36,35 @@ impl ChainInfo {
     ) -> anyhow::Result<Option<Self>> {
         let registrations =
             last_registration_chain(persistent_session, volatile_session, catalyst_id).await?;
-        if registrations.is_empty() {
-            return Ok(None);
-        }
 
         let network = Settings::cardano_network();
 
         let mut last_persistent_txn = None;
         let mut last_volatile_txn = None;
         let mut last_persistent_slot = 0.into();
 
-        let mut iter = registrations.into_iter();
-        let (is_persistent, reg) = iter.next().context("No root registration found")?;
-        let cip509 = registration(network, reg.slot_no.into(), reg.txn_index.into()).await?;
-        let mut chain = RegistrationChain::new(cip509).context("Invalid root registration")?;
-        update_values(
-            is_persistent,
-            &reg,
-            &mut last_persistent_txn,
-            &mut last_volatile_txn,
-            &mut last_persistent_slot,
-        );
-
-        for (is_persistent, reg) in iter {
-            let cip509 = registration(network, reg.slot_no.into(), reg.txn_index.into()).await?;
-
-            // This isn't a hard error because while the individual registration can be valid it can
-            // be invalid in the context of the whole registration chain.
-            if let Ok(new) = chain.update(cip509) {
-                chain = new;
+        let reqs_iter = registrations
+            .into_iter()
+            .inspect(|(is_persistent, reg)| {
```

Comment:
Let's say we have `A`, `B` and `C` persistent registrations and `D` and `E` volatile ones. Because we are querying them in parallel we can potentially query `D` and `E` then a new snapshot is accepted and the request from the persistent database returns `A, B, C, D`. After we combine both lists we get `A, B, C, D, D, E`. Can this happen? I'm not very familiar with snapshot processing.

The duplication itself shouldn't be a problem because we just discard a transaction that cannot be added to the chain. But if this situation can happen then we can inaccurately report last persistent and volatile transactions.

Reply:
>After we combine both lists we get A, B, C, D, D, E.
Yes you are right, this scenario could happen.
>But if this situation can happen then we can inaccurately report last persistent and volatile transactions.
Need to think... `last_persistent_slot` and `last_persistent_txn` will be calculated correctly anyway.